### PR TITLE
Fix issues/pr list broken when there are many repositories

### LINKED ISSUE #7750: Error 500 when visit /issues and /pulls (too many SQL variables)
- Gitea version (or commit ref): 1.9.0 / 76408d50fb
- Git version: 2.20.1
- Operating system: macOS
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist: 
```
routers/user/home.go:269:Issues() [E] CountIssuesByRepo: too many SQL variables
```
## Description

The above error should be able to explain this issue.

https://github.com/go-gitea/gitea/blob/cd238bc41598e69a78b4d17370c8929911c5f249/models/issue.go#L1357-L1366

https://github.com/go-gitea/gitea/blob/cd238bc41598e69a78b4d17370c8929911c5f249/models/issue.go#L1649-L1658

https://github.com/go-gitea/gitea/blob/cd238bc41598e69a78b4d17370c8929911c5f249/models/issue.go#L1708-L1710

### PR DESCRIPTION
**Pull request by lunny**
Fix #7750

This PR adds a new repositories sub query as condition instead of list all repository ids.

### CODE REVIEW

File: routers/user/home.go [hunk 1]
```diff
@@ -216,22 +222,12 @@ func Issues(ctx *context.Context) {
 		if isPullList {
 			unitType = models.UnitTypePullRequests
 		}
-		userRepoIDs, err = ctxUser.GetAccessRepoIDs(unitType)
-		if err != nil {
-			ctx.ServerError("ctxUser.GetAccessRepoIDs", err)
-			return
-		}
+		opts.RepoSubQuery = ctxUser.UnitRepositoriesSubQuery(unitType)
```
Comment (guillep2k) [inline routers/user/home.go]: Later on `opts.RepoIDs` is overwritten in several occasions. `RepoSubQuery` needs to be cleared in some of those cases. For example in:
```
239:		// force an empty result
240:		opts.RepoIDs = []int64{-1}
```

**Review by guillep2k — CHANGES_REQUESTED** [review]

File: routers/user/home.go (hunk 1 above)
Comment (lunny) [reply routers/user/home.go]: I fixed this one with a bigger change.

**Review by guillep2k — APPROVED** [review]

**Comment by guillep2k** [comment]
Sorry, I've missed that when reviewing.

File: models/issue.go [hunk 2]
```diff
@@ -1652,6 +1652,8 @@ func GetUserIssueStats(opts UserIssueStatsOptions) (*IssueStats, error) {
 	var repoCond = builder.NewCond()
 	if opts.RepoSubQuery != nil {
 		repoCond = builder.In("issue.repo_id", opts.RepoSubQuery)
+	} else {
+		repoCond = builder.Expr("0=1")
 	}
```
Comment (guillep2k) [inline models/issue.go]:
```
	var repoCond = builder.NewCond()
	if opts.RepoSubQuery != nil {
		repoCond = builder.In("issue.repo_id", opts.RepoSubQuery)
	} else {
		repoCond = builder.Expr("0=1")
	}
```
I'd move this block inside the `switch/case FilterModeAll:`. Having it here is confusing.

Comment (lunny) [reply models/issue.go]: But `stats.YourRepositoriesCount` on line 1742 will also use that.

Comment (guillep2k) [reply models/issue.go]: Sorry, I missed that.

**Review by techknowlogick — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
@lunny ready

**Pull request merged**